Rename `example` to `test case` in many places (#4631)
HypothesisWorks/hypothesis: hypothesis/docs/how-to/type-strategies.rst

    reveal_type(st.lists(st.integers()))
    # SearchStrategy[list[int]]

> |SearchStrategy| is the type of a strategy. It is parametrized by the type of the values it generates. You can use it to write type hints for your functions which return a strategy:

Reviewer: ```suggestion
|SearchStrategy| is the type of a strategy. It is [generic](https://typing.python.org/en/latest/spec/generics.html) in the type of the values it generates. You can use it to write type hints for your functions which return a strategy:
```

We could also say that `SearchStrategy[int]` is treated as a subtype of `SearchStrategy[object]` (ie it's covariant) because this usually helps with typechecking, even though it's not always strictly true at runtime.  Probably not worth it in a howto guide though...
